### PR TITLE
Add support for subscription with label selector

### PR DESCRIPTION
**Pull request by Ralls0**
# Description

Add support for subscription with label selector

### CODE REVIEW

**Review by giorio94 — APPROVED** [review]
/lgtm

@GabriFila Could you please take a look before merging?

File: qlkube/src/decorateSubscription.js
Comment (QcFe) [inline qlkube/src/decorateSubscription.js:58]:
```suggestion
function getResourceApiMainType(fieldName) {
  return subscriptions.find(({ type })=> `${type}Update` === fieldName);
}
```

File: qlkube/src/decorateSubscription.js
Comment (QcFe) [inline qlkube/src/decorateSubscription.js:32]:
```suggestion
    .map(({ parents, type }) => parents.includes(targetType) && type)
    .filter(s =>s);
```

File: qlkube/src/decorateSubscription.js
Comment (QcFe) [inline qlkube/src/decorateSubscription.js:45]:
```suggestion
    .map(s =>uncapitalizeType(s));
```

File: qlkube/src/decorateSubscription.js
Comment (QcFe) [inline qlkube/src/decorateSubscription.js:102]: I would suggest making a small function (with lots of returns) to improve readability, also it could be recycled afterwards

File: qlkube/src/decorateSubscription.js
Comment (QcFe) [inline qlkube/src/decorateSubscription.js:332]: Would make a variable earlier in order to make this cleaner

File: qlkube/src/decorateSubscription.js
Comment (QcFe) [inline qlkube/src/decorateSubscription.js:435]: As before, I'd extract this into a function

**Review by QcFe — COMMENTED** [review]
Some minor style comments

File: qlkube/src/decorateSubscription.js
Comment (Ralls0) [reply qlkube/src/decorateSubscription.js:32]: So, I think that even `if (wtype['parents'].includes(targetType)) return wtype['fieldWrapper'];` could be changed at the same way

File: qlkube/src/decorateSubscription.js
Comment (Ralls0) [reply qlkube/src/decorateSubscription.js:332]: like: 

`const isTenant = resourceApiMainType.resource === 'tenants'` 

and then:

```
name: ${
          isTenant  ? 'String!' : 'String'
```

or somethings like:

`nameType = resourceApiMainType.resource === 'tenants' ? 'String!' : 'String';`

and then:

`name: ${nameType}`

File: qlkube/src/decorateSubscription.js
Comment (QcFe) [reply qlkube/src/decorateSubscription.js:32]: yes, I wrote it once but it can clearly be applied everywhere. Just test it behaves correctly because you can never know what will happen with js 🥴

File: qlkube/src/decorateSubscription.js
Comment (QcFe) [reply qlkube/src/decorateSubscription.js:332]: the second you wrote

**Comment by giorio94** [comment]
/hold

The deploy-staging is crashing

**Pull request merged**
